src/dfs.ts

```diff
@@ -0,0 +1,245 @@
+    createFeeVaultParams: CreateFeeVaultParams
+  ): Promise<Transaction> {
+    const { feeVault, tokenMint, owner, payer, userShare } =
+      createFeeVaultParams;
+
+    const params: InitializeFeeVaultParameters = {
+      padding: [],
+      users: userShare.map((share) => ({
+        address: share.address,
+        share: share.share,
+      })),
+    };
+
+    const tokenVault = deriveTokenVaultAddress(feeVault);
+
+    const tokenProgram = await getTokenProgram(tokenMint, this.connection);
+
+    return this.program.methods
+      .initializeFeeVault(params)
+      .accountsPartial({
+        feeVault,
+        feeVaultAuthority: this.feeVaultAuthority,
+        tokenVault,
+        tokenMint,
+        owner,
+        payer,
+        tokenProgram,
+      })
+      .transaction();
+  }
+
+  /**
+   * Create a fee vault PDA
+   * @param createFeeVaultPdaParams - The parameters for creating a fee vault PDA
+   * @returns The transaction to create a fee vault PDA
+   */
+  async createFeeVaultPda(
+    createFeeVaultPdaParams: CreateFeeVaultPdaParams
+  ): Promise<Transaction> {
+    const { base, tokenMint, owner, payer, userShare } =
+      createFeeVaultPdaParams;
+
+    const params: InitializeFeeVaultParameters = {
+      padding: [],
+      users: userShare.map((share) => ({
+        address: share.address,
+        share: share.share,
+      })),
+    };
+
+    const feeVault = deriveFeeVaultPdaAddress(base, tokenMint);
+    const tokenVault = deriveTokenVaultAddress(feeVault);
+
+    const tokenProgram = await getTokenProgram(tokenMint, this.connection);
+
+    return this.program.methods
+      .initializeFeeVaultPda(params)
+      .accountsPartial({
+        feeVault,
+        base,
+        feeVaultAuthority: this.feeVaultAuthority,
+        tokenVault,
+        tokenMint,
+        owner,
+        payer,
+        tokenProgram,
+      })
+      .transaction();
+  }
+
+  /**
+   * Fund a fee vault
+   * @param fundFeeVaultParams - The parameters for funding a fee vault
+   * @returns The transaction to fund a fee vault
+   */
+  async fundFeeVault(
+    fundFeeVaultParams: FundFeeVaultParams
+  ): Promise<Transaction> {
+    const { fundAmount, feeVault, funder } = fundFeeVaultParams;
+
+    const feeVaultState = await this.getFeeVault(feeVault);
+    const tokenVault = feeVaultState.tokenVault;
+    const tokenMint = feeVaultState.tokenMint;
+
+    const tokenProgram = await getTokenProgram(tokenMint, this.connection);
+
+    const { ataPubkey: fundTokenVault, ix: preInstruction } =
+      await getOrCreateATAInstruction(
+        this.program.provider.connection,
+        tokenMint,
+        funder,
+        funder,
+        true,
+        tokenProgram
+      );
+
+    const txBuilder = this.program.methods.fundFee(fundAmount).accountsPartial({
+      feeVault,
+      tokenVault,
+      tokenMint,
+      fundTokenVault,
+      funder,
+      tokenProgram,
+    });
+
+    const preInstructions = [];
+
+    if (preInstruction) {
```

Comment:
i think you should move logic prepare pre-post instruction to above txBuilder will more clean.
then you can use like this. and no need to to more if-else check
```ts
tsBuilder = program.methods.fundFee(amount).account({...}).preInstructions(...).postInstructions(...)
```